### PR TITLE
Fix issue #1898: OData.Core lib fails to parse IN operator with Date, DateTimeOffset and Time parameters.

### PR DESCRIPTION
**Pull request by maxbruecken**
Fix parsing of IN operator with Date, DateTimeOffset, Time and Duration parameters.

### Issues

*This pull request fixes issue #1898 *

### Description

Date, DateTimeOffset, Time and Duration values should be single-quoted to be compatible with Json reader.

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by ElizabethOkerio — COMMENTED** [review]
@maxbruecken can you add test cases for parsing filters with null datetime, datetimeoffset and date for nullable datetime, date and datetimeoffset. 

For instance if I have such a model:
```csharp
public class TestEntity
{
   public DateTime? NullableDate { get; set; }
}
```
the following URL should be parsed successfully
```url
http://host/service/TestEntity?$filter=NullableDate in (null)
```

**Comment by maxbruecken** [comment]
> @maxbruecken can you add test cases for parsing filters with null Date, DateTimeOffset and Time etc for nullable Date, DateTimeOffset and Time .
> 
> For instance if I have such a model:
> 
> ```cs
> public class TestEntity
> {
>    public Date? NullableDate { get; set; }
> }
> ```
> 
> the following URL should be parsed successfully
> 
> ```
> http://host/service/TestEntity?$filter=NullableDate in (null)
> ```

Good point, thank you. I have added the properly handling of nulls in my last commit.

File: src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs [hunk 1]
```diff
@@ -313,5 +326,30 @@ private static string NormalizeGuidCollectionItems(string bracketLiteralText)
 
             return "[" + String.Join(",", items) + "]";
         }
+
+        private static string NormalizeDateTimeCollectionItems(string bracketLiteralText)
```
Comment (KenitoInc) [inline src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs:330]: To ensure there are no bugs in this method, add a few tests with one parameter
e.g
`FilterClause filter = ParseFilter("Birthdate in (1950-01-02T06:15:00Z)",
                HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType());`

`FilterClause filter = ParseFilter("Birthdate in (null)",
                HardCodedTestModel.TestModel, HardCodedTestModel.GetPersonType());`

**Review by KenitoInc — CHANGES_REQUESTED** [review]

File: src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs (hunk 1 above)
Comment (maxbruecken) [reply src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs:330]: I have added two tests with date/time collections containing single item or single null

**Review by KenitoInc — APPROVED** [review]
LGTM

**Review by ElizabethOkerio — APPROVED** [review]

**Review by mikepizzo — APPROVED** [review]
:shipit:
